src/model/event.ts

```diff
@@ -403,6 +403,48 @@ export class LearnEffect extends Effect {
     }
 }
 
+/**
+ *  An effect that selects the an unlearned learnable from the given collection.
+ *  LearnNextEffect needs to be initialized with a store before it can be used.
+ */
+export class LearnNextEffect extends LearnEffect {
+    // TODO: this is not used atm
+    collection: model.CollectionId;
+
+    constructor(collection: model.CollectionId, customLogMessage?: string) {
+        super("", customLogMessage);
+        this.collection = collection;
+    }
+
+    init(store: model.Store) {
```

Comment:
This is a little janky, though I see why you did it (gotta share state between action/event log).